src/main/java/frc/robot/subsystems/drivetrain/tests/DriveMotorsSpeedTest.java

```diff
@@ -0,0 +1,57 @@
+package frc.robot.subsystems.drivetrain.tests;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.drivetrain.SwerveDrive;
+
+public class DriveMotorsSpeedTest extends CommandBase {
+    private final SwerveDrive swerveDrive;
+    private final Timer timer = new Timer();
+    double times[] = new double[4];
+
+    public DriveMotorsSpeedTest(SwerveDrive swerveDrive) {
+        this.swerveDrive = swerveDrive;
+        addRequirements(swerveDrive);
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+        timer.start();
+    }
+
+    @Override
+    public void execute() {
+        swerveDrive.setStates(new SwerveModuleState[]{
+                new SwerveModuleState(3, Rotation2d.fromDegrees(0)),
+                new SwerveModuleState(3, Rotation2d.fromDegrees(0)),
+                new SwerveModuleState(3, Rotation2d.fromDegrees(0)),
+                new SwerveModuleState(3, Rotation2d.fromDegrees(0))
+        });
+        for (int i = 0; i < 4; i++) {
+            if (Math.abs(Math.abs(swerveDrive.getModule(i).getVelocity()) - 3) < 0.1) {
```

Comment:
You don't need the inner `abs` function if you have it on the outside.
```suggestion
            if (Math.abs(swerveDrive.getModule(i).getVelocity() - 3) < 0.1) {
```